### ISSUE TITLE
Creating transformers ... "@@"??

### ISSUE THREAD

**Issue by shaunc** [cognitect-labs/transducers-js#44]
Does the use of "@@" in the protocol mean that in es6 code I should actually do something like the following to define a transformer?
```javascript
const init = Symbol.for('transducer/init');
const result = Symbol.for('transducer/result');
const step = Symbol.for('transducer/step');
function MapTransformer() {
  return {
    [init]() { return xf[init](); }
    [result](res) { return xf[result](res); }
    [step](res, input) { return xf[step](res, f(input)); }
  };
}
```
I'm confused what the "@@" are supposed to represent as it doesn't look like you have es6 support -- I don't suppose the above is actually interoperable with the library. But isn't "@@" for "well-known symbols" in es6?

**Comment by seanpoulter** [comment]
> isn't "@@" for "well-known symbols" in es6?

Yep. Have a look at the Symbols chapter of You Don't Know JS: ES 6 & Beyond. Here's a note from the [Built-in Symbols](https://github.com/getify/You-Dont-Know-JS/blob/master/es6%20%26%20beyond/ch2.md#built-in-symbols) section:
> The specification uses the @@ prefix notation to refer to the built-in symbols, the most common ones being: `@@iterator`, `@@toStringTag`, `@@toPrimitive`.

<br/>

That leads to the question, are a String and a Symbol equivalent? You can use the following block to spot check if they're the same. It looks to me like you'll need a PR to support Symbols before your code with Symbols.
```
Array.prototype['@@custom'] = 'from String';
console.log([]['@@custom']);
// > 'value'

const sym = Symbol.for('custom');
console.log([][sym]);
// > undefined

Array.prototype[sym] = 'from Symbol'
console.log([][sym]);
// > 'from Symbol'
```
